File: apps/clinical-engine/src/ehas2_clinical_engine/rule4/phase/day_bands.py
from __future__ import annotations


def fallback_phase_from_duration_days(days: int) -> str:
    if 1 <= days <= 14:
        return "ACUTE"
    if 15 <= days <= 45:
        return "SUB_ACUTE"
    if 46 <= days <= 90:
        return "CHRONIC_MODERATE"
    return "DEEP_CHRONIC"

File: packages/clinical-contracts/src/rule4/phase/dayBands.ts
import type { Rule4ClinicalPhase } from './types.js';

export function fallbackPhaseFromDurationDays(days: number): Rule4ClinicalPhase {
  if (days >= 1 && days <= 14) {
    return 'ACUTE';
  }
  if (days >= 15 && days <= 45) {
    return 'SUB_ACUTE';
  }
  if (days >= 46 && days <= 90) {
    return 'CHRONIC_MODERATE';
  }
  return 'DEEP_CHRONIC';
}
